maven-repository-metadata/src/main/mdo/metadata.mdo

```diff
@@ -316,26 +348,30 @@ under the License.
           <name>classifier</name>
           <version>1.1.0+</version>
           <type>String</type>
-          <description>The classifier of the sub-artifact.</description>
+          <description>The classifier of the sub-artifact. Each classifier and extension pair may only appear once.</description>
           <defaultValue></defaultValue>
+          <identifier>true</identifier>
         </field>
         <field>
           <name>extension</name>
           <version>1.1.0+</version>
           <type>String</type>
-          <description>The file extension of the sub-artifact.</description>
+          <description>The file extension of the sub-artifact. Each classifier and extension pair may only appear once.</description>
+          <identifier>true</identifier>
         </field>
         <field xml.tagName="value">
           <name>version</name>
           <version>1.1.0+</version>
           <type>String</type>
           <description>The resolved snapshot version of the sub-artifact.</description>
+          <identifier>true</identifier>
         </field>
         <field>
           <name>updated</name>
           <version>1.1.0+</version>
           <type>String</type>
           <description>The timestamp when this version information was last updated. The timestamp is expressed using UTC in the format yyyyMMddHHmmss.</description>
+          <identifier>true</identifier>
```

Comment:
Can you show where `version` and `updated` are part of an id now?

Reply:
This is for letting Modello generate a complete `equals` method which is used e.g. in https://github.com/apache/maven/pull/681/files#diff-a8781d28a4f703eaad1745f9002678d2e273d9189fb9a5f97a79df346f4c1953R129